gitlab/client_repository_teamaccess.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitlab
+
+import (
+	"context"
+	"errors"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+)
+
+// TeamAccessClient implements the gitprovider.TeamAccessClient interface.
+var _ gitprovider.TeamAccessClient = &TeamAccessClient{}
+
+// TeamAccessClient operates on the teams list for a specific repository.
+type TeamAccessClient struct {
+	*clientContext
+	ref gitprovider.RepositoryRef
+}
+
+// Get a team within the specific organization.
+//
+// name may include slashes, but must not be an empty string.
+// Teams are sub-groups in GitLab.
+//
+// ErrNotFound is returned if the resource does not exist.
+func (c *TeamAccessClient) Get(ctx context.Context, teamName string) (gitprovider.TeamAccess, error) {
+	// project, err := c.c.GetGroupProject(ctx, groupName, c.ref.GetRepository())
+	project, err := c.c.GetGroupProject(ctx, c.ref.GetIdentity(), c.ref.GetRepository())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, group := range project.SharedWithGroups {
+		if group.GroupName == teamName {
+			gitProviderPermission, err := getGitProviderPermission(group.GroupAccessLevel)
+			if err != nil {
+				return nil, err
+			}
+
+			return newTeamAccess(c, gitprovider.TeamAccessInfo{
+				Name:       teamName,
+				Permission: gitProviderPermission,
+			}), nil
+		}
+	}
```

Comment:
Could pull this out into a similar fn to the github `GetTeamPermissions` to keep things more symmetrical but also it doesn't seem to be used anywhere else really so AG 👌

Reply:
Was looking for this as well but it doesn't seem to be possible from the Gitlab API directly... Seems like only the way to list them is by `project.SharedWithGroups`